storage/store.go

```diff
@@ -1533,19 +1533,39 @@ func (s *store) putLayer(rlstore rwLayerStore, rlstores []roLayerStore, id, pare
 			GIDMap:         copySlicePreferringNil(gidMap),
 		}
 	}
-	return rlstore.create(id, parentLayer, names, mountLabel, nil, &options, writeable, diff, slo)
+	return rlstore.create(id, parentLayer, names, mountLabel, nil, &options, writeable, slo)
 }
 
 func (s *store) PutLayer(id, parent string, names []string, mountLabel string, writeable bool, lOptions *LayerOptions, diff io.Reader) (*Layer, int64, error) {
 	rlstore, rlstores, err := s.bothLayerStoreKinds()
 	if err != nil {
 		return nil, -1, err
 	}
+
+	var slo *stagedLayerOptions
+
+	if diff != nil {
+		m := newMaybeStagedLayerExtraction(diff, s.graphDriver)
```

Comment:
Formally, this is racy WRT updates of `s.graphDriver` … while using `rlstore.driver` is not.